src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -1093,6 +1094,14 @@ static ErrorCode getRefMismatchErrorCode(TypeKind type)
             MethodSymbol selectedMethod = conversion.Method;
 
             var location = syntax.Location;
+
+            if (!isAddressOf)
+            {
+                ReportDiagnosticsIfUnmanagedCallersOnly(diagnostics, selectedMethod, location, isDelegateConversion: true);
```

Comment:
>ReportDiagnosticsIfUnmanagedCallersOnly [](start = 16, length = 39)

If we're moving the call to `ReportDiagnosticsIfUnmanagedCallersOnly()`, we should have a test that covers the change in behavior.

Reply:
An example test:
```C#
using System.Runtime.InteropServices;
unsafe delegate byte* D();
class C
{
    [UnmanagedCallersOnly]
    unsafe static byte* F() => default;
    unsafe static D M1() => new D(F);
    static D M2() => new D(F);
}
```